up_client_socket/java/src/main/java/org/eclipse/uprotocol/SocketUTransport.java

```diff
@@ -1,274 +1,241 @@
+      *
+      * @param sourceFilter The URI filter for the source.
+      * @param sinkFilter The URI filter for the sink.
+      * @param listener The listener to be registered.
+      */
+     public void addListener(UUri sourceFilter, UUri sinkFilter, UListener listener) {
+         logger.info("listeners: " + sourceFilter + ", " + sinkFilter + ", " + listener);
+         AbstractMap.SimpleEntry<UUri, UUri> key = new AbstractMap.SimpleEntry<>(sourceFilter, sinkFilter);
+         uriToListener.put(key, listener);
+     }
+ 
+     /**
+      * Removes the listener registered for the specified source and sink URI filters.
+      *
+      * @param sourceFilter The URI filter for the source.
+      * @param sinkFilter The URI filter for the sink.
+      * @return A status indicating the outcome of the removal operation.
+      */
+     public UStatus removeListener(UUri sourceFilter, UUri sinkFilter) {
+         AbstractMap.SimpleEntry<UUri, UUri> key = new AbstractMap.SimpleEntry<>(sourceFilter, sinkFilter);
+         UListener listener = uriToListener.remove(key);
+         if (listener != null) {
+             return UStatus.newBuilder().setCode(UCode.OK).setMessage("Listener removed successfully").build();
+ 
+         } else {
+             return UStatus.newBuilder().setCode(UCode.NOT_FOUND).setMessage("Listener not found for the given URI")
+                     .build();
+         }
+     }
+ 
+     /**
+      * Sends the provided message over the socket connection.
+      *
+      * @param message The message to be sent.
+      * @return A status indicating the outcome of the send operation.
+      */
+     public CompletionStage<UStatus> send(UMessage message) {
+         byte[] umsgSerialized = message.toByteArray();
+         try {
+             OutputStream outputStream = socket.getOutputStream();
+             outputStream.write(umsgSerialized);
+             logger.info("uMessage Sent to dispatcher fron java socket transport");
+             return CompletableFuture.completedFuture(
+                     UStatus.newBuilder().setCode(UCode.OK).setMessage("uMessage Sent to dispatcher").build());
+         } catch (IOException e) {
+             logger.log(Level.SEVERE, INTERNAL_ERROR, e);
+             return CompletableFuture.completedFuture(
+                     UStatus.newBuilder().setCode(UCode.INTERNAL).setMessage(INTERNAL_ERROR + e.getMessage()).build());
+         }
+     }
+ 
+     /**
+      * Registers the specified listener for the given source and sink URI filters.
+      *
+      * @param sourceFilter The URI filter for the source.
+      * @param sinkFilter The URI filter for the sink.
+      * @param listener The listener to be registered.
+      * @return A status indicating the outcome of the register listener operation.
+      */
+     public CompletionStage<UStatus> registerListener(UUri sourceFilter, UUri sinkFilter, UListener listener) {
```

Comment:
sink uri can be optional too. Why isn't that API covered ?

https://github.com/eclipse-uprotocol/up-java/blob/main/src/main/java/org/eclipse/uprotocol/transport/UTransport.java

Reply:
default CompletionStage<UStatus> registerListener(UUri sourceFilter, UListener listener) {
        return registerListener(sourceFilter, UriFactory.ANY, listener);
    }
    
    This default from up-java will handle that case.